src/Rocketlet.ts

```diff
@@ -78,12 +82,18 @@ export abstract class Rocketlet {
      *
      * @return whether the Rocketlet should be enabled or not
      */
-    public abstract onEnable(): boolean;
+    public abstract onEnable(environment: IEnvironmentRead, configurationModify: object): boolean; //todo: Config modify. This should actually be an implementation of configModify which ensures that only own configurations are being modified
```

Comment:
The signature documents what's expected to be done in that method. 
In this case, in initialization the configuration can be extended while the actual modification with values would be done in enable - to reset it in disable. 
In enable and in disable, the accessor should be implemented in such a way that it allows modification of own settings, slashCommands etc.
In a dedicated other method, disruptive changes could be done (e. g. hiding default settings)
Feels good to me.